include/netdb.h

```diff
@@ -156,6 +156,28 @@
 #define NO_RECOVERY    3
 #define TRY_AGAIN      4
 
+/* NI_MAXHOST is the max of
+ *
+ *    CONFIG_NETDB_DNSCLIENT_NAMESIZE + 1
+ *    INET6_ADDRSTRLEN
+ *    INET_ADDRSTRLEN
+ *
+ * Note: INETxxx_ADDRSTRLEN already includes the terminating NUL.
+ * Note: INET6_ADDRSTRLEN > INET_ADDRSTRLEN is assumed.
+ */
+
+#if CONFIG_NETDB_DNSCLIENT_NAMESIZE > INET6_ADDRSTRLEN
+#define	NI_MAXHOST	CONFIG_NETDB_DNSCLIENT_NAMESIZE
+#else
+#define	NI_MAXHOST	INET6_ADDRSTRLEN
```

Comment:
should we check CONFIG_NET_IPv6 and CONFIG_NET_IPv4 here?

Reply:
> should we check CONFIG_NET_IPv6 and CONFIG_NET_IPv4 here?

maybe.

> And should we replace CONFIG_NETDB_DNSCLIENT_NAMESIZE in source code with NI_MAXHOST? since NI_MAXHOST may different from CONFIG_NETDB_DNSCLIENT_NAMESIZE.

it depends.

Reply:
i fixed config checks here